ext/drbbdup/drbbdup.c

```diff
@@ -751,31 +776,115 @@ drbbdup_encode_runtime_case(void *drcontext, drbbdup_per_thread *pt, void *tag,
 
     /* Load the encoding to the scratch register. */
     /* FIXME i#4134: Perform lock if opts.atomic_load_encoding is set. */
+    opnd_t case_opnd = opts.runtime_case_opnd;
     opnd_t scratch_reg_opnd = opnd_create_reg(DRBBDUP_SCRATCH_REG);
-    instr_t *instr =
-        XINST_CREATE_load(drcontext, scratch_reg_opnd, opts.runtime_case_opnd);
+#ifdef AARCHXX
+    if (opnd_is_rel_addr(case_opnd)) {
+        /* Work around two problems:
+         * 1) DR's AArch64 decoder doesn't yet support OP_ldr with pc-rel opnd
+         *    (i#4847, i#5316).
+         * 2) To ensure we can reach we may need to load the address into the
+         *    register in a separate step.  DR may mangle this for us (i#1834)
+         *    so we may be able to remove this in the future.
+         */
+        instrlist_insert_mov_immed_ptrsz(drcontext, (ptr_int_t)opnd_get_addr(case_opnd),
+                                         scratch_reg_opnd, bb, where, NULL, NULL);
+        case_opnd = OPND_CREATE_MEMPTR(DRBBDUP_SCRATCH_REG, 0);
+    }
+#endif
+    instr_t *instr = XINST_CREATE_load(drcontext, scratch_reg_opnd, case_opnd);
     instrlist_meta_preinsert(bb, where, instr);
 }
 
-#ifdef X86_64
 static void
-drbbdup_insert_compare_encoding(void *drcontext, instrlist_t *bb, instr_t *where,
-                                drbbdup_case_t *current_case, reg_id_t reg_encoding)
+drbbdup_insert_compare_encoding_and_branch(void *drcontext, instrlist_t *bb,
+                                           instr_t *where, drbbdup_case_t *current_case,
+                                           reg_id_t reg_encoding, bool jmp_if_equal,
+                                           instr_t *jmp_label)
 {
+#define MINSERT instrlist_meta_preinsert
```

Comment:
Nit: (me being super picky here!!) It seems odd, in terms of style, that this definition is here and not universally used throughout the file. I would remove the definition in this PR, and then have another PR that solely introduces `MINSERT` and replaces all occurrences of  `instrlist_meta_preinsert` with the definition in  drbbdup.c.

Reply:
That's fair: consistency is good.  Done.  (Agreed that macros are generally not preferred, but with these long lines and with more and more instructions being inserted it can help to make them more concise and get closer to an assembly listing (xref #4510).)